gnucash/gnome-search/search-core-type.c

```diff
@@ -230,7 +230,9 @@ editable_enters (GNCSearchCoreType *fe)
 void
 gnc_search_core_register_type (const char *type_name, GNCSearchCoreNew fcn)
 {
-    g_return_if_fail (type_name || *type_name || fcn);
+    g_return_if_fail (type_name);
+    g_return_if_fail (*type_name);
+    g_return_if_fail (fcn);
```

Comment:
Just change the `||` to `&&`.

Reply:
Yes, that was my first instinct, but it's annoyingly inconsistent with the next line.

Reply:
Not really, but maybe it's because you don't know that `g_return_if_fail` also logs an error. This one ensures that the function args precondition is met, the other one ensures that the static variable `typeTable` has been initialized. Two quite different preconditions that deserve separate error messages.